contactaccessrights/CRM/Contactaccessrights/Helper/ContactActionsMenu/ContactAccessActionGroup.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+use CRM_HRContactActionsMenu_Component_Group as ActionsGroup;
+use CRM_Contactaccessrights_Service_ContactRights as ContactRightsService;
+use CRM_HRCore_CMSData_UserPermissionInterface as CMSUserPermission;
+use CRM_HRContactActionsMenu_Component_GroupButtonItem as ActionsGroupButtonItem;
+use CRM_Contactaccessrights_Component_ContactActionsMenu_GroupTitleToolTipItem as GroupTitleToolTipItem;
+use CRM_Contactaccessrights_Component_ContactActionsMenu_NoStaffTooltipItem as NoStaffToolTipItem;
+use CRM_Contactaccessrights_Component_ContactActionsMenu_GenericTextItem as GenericTextItem;
+use CRM_Contactaccessrights_Component_ContactActionsMenu_UserRegionsListItem as UserRegionsListItem;
+use CRM_Contactaccessrights_Component_ContactActionsMenu_UserLocationsListItem as UserLocationsListItem;
+use CRM_Contactaccessrights_Component_ContactActionsMenu_UserACLGroupsListItem as UserACLGroupsListItem;
+
+class CRM_Contactaccessrights_Helper_ContactActionsMenu_ContactAccessActionGroup {
+
+  /**
+   * @var array
+   */
+  private $contactUserInfo;
+
+  /**
+   * @var ContactRightsService
+   */
+  private $contactRightsService;
+
+  /**
+   * @var array
+   */
+  private $contactACLGroups;
+
+  /**
+   * @var CMSUserPermission
+   */
+  private $CMSUserPermission;
+
+  /**
+   * CRM_Contactaccessrights_Helper_ContactActionsMenu_ContactAccessActionGroup constructor.
+   *
+   * @param array $contactUserInfo
+   * @param ContactRightsService $contactRightsService
+   * @param CMSUserPermission $CMSUserPermission
+   * @param array $contactACLGroups
+   */
+  public function __construct(
+    $contactUserInfo,
+    ContactRightsService $contactRightsService,
+    CMSUserPermission $CMSUserPermission,
+    $contactACLGroups
+  ) {
+    $this->contactUserInfo = $contactUserInfo;
+    $this->contactRightsService = $contactRightsService;
+    $this->CMSUserPermission = $CMSUserPermission;
+    $this->contactACLGroups = $contactACLGroups;
+  }
+
+  /**
+   * Gets Contact Access Menu Group with menu items already
+   * added.
+   *
+   * @return ActionsGroup
+   */
+  public function get() {
+    $actionsGroup = new ActionsGroup($this->getGroupTitle());
+    $isAdmin = $this->isAdmin();
+
+    if ($isAdmin) {
+      $allStaffItem =  new GenericTextItem('All Staff');
+      $actionsGroup->addItem($allStaffItem);
+    }
+
+    if (!$isAdmin) {
+      $regions = $this->contactRightsService->getContactRightsByRegions($this->contactUserInfo['contact_id']);
```

Comment:
This is a bit too long. the line can be shortened if you create a method like `getContactRegions()` that, internally, call the Contact Rights Service method. The same can be done for the locations. To make your life easier when create the list items, these new methods could return only a list of regions/locations labels, since that's what you really need here